### ISSUE TITLE
BatchNormalization values wrong?

### ISSUE THREAD

**Issue by christian-vorhemus** [microsoft/CNTK#3672]
I've trained a model in CNTK with the following layers:

```python
from cntk.layers import Dense, Sequential, Activation, Convolution2D, MaxPooling, Dropout, BatchNormalization
from cntk import softmax, relu

model = Sequential([
    Convolution2D(filter_shape=(5,5), num_filters=32, strides=(1,1), pad=True, name="first_conv"),
    BatchNormalization(),
    Activation(relu),
    MaxPooling(filter_shape=(3,3), strides=(2,2), name="first_max"),
    Convolution2D(filter_shape=(3,3), num_filters=48, strides=(1,1), pad=True, name="second_conv"),
    BatchNormalization(),
    Activation(relu),
    MaxPooling(filter_shape=(3,3), strides=(2,2), name="second_max"),
    Convolution2D(filter_shape=(3,3), num_filters=64, strides=(1,1), pad=True, name="third_conv"),
    BatchNormalization(),
    Activation(relu),
    MaxPooling(filter_shape=(3,3), strides=(2,2), name="third_max"),
    Convolution2D(filter_shape=(5,5), num_filters=32, strides=(1,1), pad=True, name="fourth_conv"),
    Activation(relu),
    Dense(100, activation=relu),
    Dropout(0.1),
    Dense(3, activation=softmax)
])

```
I serialized it with
```python
model.save("model.onnx", format=ModelFormat.ONNX)
```

When I visualize the layers, the BatchNormalization parameters don't look right, I also can't run this model in ONNX runtime, getting the error message that the scale value dimension in BatchNormalization is wrong:

![aaa](https://user-images.githubusercontent.com/3229424/58721395-301d2580-83d5-11e9-99e7-f9f6d9710a02.png)

For comparison, when I create the following model in Keras:

```python
from keras.models import Sequential
from keras.layers import Dense, Conv2D, MaxPool2D, Dropout, Flatten, BatchNormalization, InputLayer

model = Sequential([
    InputLayer(input_shape=[3,168,224]),
    Conv2D(filters=32, kernel_size=5, strides=1, padding='same'),
    BatchNormalization(),
    Activation('relu'),
    MaxPool2D(pool_size=8, padding='same'),
    Conv2D(filters=48, kernel_size=3, strides=1, padding='same'),
    BatchNormalization(),
    Activation('relu'),
    MaxPool2D(pool_size=5, padding='same'),
    Conv2D(filters=64, kernel_size=3, strides=1, padding='same'),
    BatchNormalization(),
    Activation('relu'),
    MaxPool2D(pool_size=3, padding='same'),
    Conv2D(filters=32, kernel_size=5, strides=1, padding='same'),
    Flatten(),
    Dense(100, activation='relu'),
    Dropout(0.1),
    Dense(3, activation='softmax')
])
```
and serialize it, the BatchNormalization parameters look like this:

![bbb](https://user-images.githubusercontent.com/3229424/58722202-24326300-83d7-11e9-95e0-f067c4d5546d.png)

Did I miss something in my CNTK graph definition?

**System information**
- OS Platform: Windows 10
- CNTK version: 2.7
- Python version: 3.6.8

**Comment by delzac** [comment]
Hi @christian-vorhemus ,

You need to set map_rank=1 like the following and it will work as intended:
`BatchNormalization(map_rank=1)`

```
map_rank (1 or ``None``): passing 1 means spatially-pooled batch-normalization, where normalization values will be tied across all pixel positions; while ``None``
      will normalize all elements of the input tensor independently
```

**Comment by christian-vorhemus** [comment]
This did the trick, thanks for the clarification @delzac !
